src/devtools/useAtomsDevtools.ts

```diff
@@ -0,0 +1,174 @@
+import { useContext, useEffect, useRef } from 'react'
+import { useAtomsSnapshot, useGotoAtomsSnapshot } from 'jotai/devtools'
+import { Atom, Scope } from '../core/atom'
+import { getScopeContext } from '../core/contexts'
+import {
+  AtomState,
+  DEV_GET_ATOM_STATE,
+  DEV_SUBSCRIBE_STATE,
+  Store,
+} from '../core/store'
+
+type Config = {
+  instanceID?: number
+  name?: string
+  serialize?: boolean
+  actionCreators?: any
+  latency?: number
+  predicate?: any
+  autoPause?: boolean
+}
+
+type Message = {
+  type: string
+  payload?: { type: string; actionId: number }
+  state?: any
+}
+
+type ConnectionResult = {
+  subscribe: (dispatch: any) => () => void
+  unsubscribe: () => void
+  send: (action: string, state: any) => void
+  init: (state: any) => void
+  error: (payload: any) => void
+}
+
+type Extension = {
+  connect: (options?: Config) => ConnectionResult
+}
+
+type AtomsSnapshot = Map<Atom<unknown>, unknown>
+
+const serializeSnapshot = (snapshot: AtomsSnapshot) => {
+  const result: Record<string, unknown> = {}
+
+  snapshot.forEach((v, atom) => {
+    result[atomToDebuggingString(atom)] = v
+  })
+
+  return result
+}
+
+const atomToDebuggingString = (atom: Atom<unknown>) =>
+  `${atom}:${atom.debugLabel ? atom.debugLabel : atom}`
+
+const getDependencies = (store: Store, snapshot: AtomsSnapshot) => {
+  const result: Record<string, string[]> = {}
+
+  snapshot.forEach((_, atom) => {
+    const atomState = store[DEV_GET_ATOM_STATE]?.(atom)
+
+    if (!atomState) {
+      return
+    }
+    result[atomToDebuggingString(atom)] = [...atomState.d.keys()].map(
+      atomToDebuggingString
+    )
+  })
+
+  return result
+}
+
+export function useAtomsDevtools(name: string, scope?: Scope) {
+  const snapshot = useAtomsSnapshot()
+  const goToSnapshot = useGotoAtomsSnapshot()
+  let extension: Extension | undefined
+  try {
+    extension = (window as any).__REDUX_DEVTOOLS_EXTENSION__ as Extension
+  } catch {}
+  if (!extension) {
+    if (
+      typeof process === 'object' &&
+      process.env.NODE_ENV === 'development' &&
+      typeof window !== 'undefined'
+    ) {
+      console.warn('Please install/enable Redux devtools extension')
+    }
+  }
+
+  const ScopeContext = getScopeContext(scope)
+  const scopeContainer = useContext(ScopeContext)
+  const store = scopeContainer.s
+
+  if (!store[DEV_SUBSCRIBE_STATE]) {
+    throw new Error('useAtomsSnapshot can only be used in dev mode.')
+  }
+
+  const isTimeTraveling = useRef(false)
+  const isRecording = useRef(true)
+  const devtools = useRef<ConnectionResult & { shouldInit?: boolean }>()
+  const snapshots = useRef<AtomsSnapshot[]>([])
+
+  useEffect(() => {
+    let devtoolsUnsubscribe: () => void | undefined
+    if (extension) {
+      devtools.current = extension.connect({ name })
```

Comment:
```suggestion
      const connection = extension.connect({ name })
      devtools.current = connection // move this to just above L134
```
can avoid L117 non-null assertion.

Reply:
i guess it didn't work